polls/impl/src/java/org/sakaiproject/poll/logic/impl/ExternalLogicImpl.java

```diff
@@ -487,26 +487,31 @@ private LRS_Statement getStatementForUserEditPoll(String text, boolean newPoll)
         return new LRS_Statement(student, verb, lrsObject);
     }
 
+    private String getPollRef(String pollId) {
+        return "poll" + getCurrentLocationReference() + "/poll/" + pollId;
+    }
+
     /**
      * @see org.sakaiproject.poll.logic.ExternalLogic#registerStatement(java.lang.String, org.sakaiproject.poll.model.Vote)
      */
     @Override
     public void registerStatement(String pollText, Vote vote) {
         if (null != learningResourceStoreService) {
```

Comment:
It feels like these two methods could be further refactored/consolidated but this is better than it was.